### ISSUE TITLE
Feature - Install Processing from debian repository

### ISSUE THREAD

**Issue by Akarshit** [processing/processing#4266]
It would be a nice feature if Processing could be directly installed from debian repositories or something similar using commands like `sudo apt-get install processing`, etc.
This would also help users to upgrade processing easily.

@shiffman Do you think this is a good feature to have ?

I would like to implement this but am having some problem as to where to start from. I will keep digging but any help would be highly appreciated.

**Comment by shiffman** [comment]
I'm not enough of a linux user to know, but I can see how it would be useful.

**Comment by aengelke** [comment]
There is support for building .deb packages using `ant deb`.

**Comment by Akarshit** [comment]
Thanks @aengelke

**Comment by benfry** [comment]
The feature is implemented (#114); it's just a matter of having a consistent maintainer.

**Comment by marado** [comment]
A first step would be what I asked in #4230 : without the overhead of maintaining a repository and all of that, just `ant deb` each time there is a new version, and add the resulting .deb next to the other files in the downloads section.

**Comment by benfry** [comment]
Same issue; needs a consistent maintainer. I'm not eager to add another step to the already long build process of releasing and testing on 5 different platforms (OS X, Windows 32 and 64, Linux 32 and 64).

**Comment by gohai** [comment]
I just tested `ant deb` on ARM w/ PR #4300, and it worked well (thanks, @kfeuz)!

I want to work towards getting a .deb into one of these Debian spin-offs for small devices, since that's where my focus is.

**Comment by benfry** [comment]
@gohai So maybe I could convince you to handle the ARM release and also the desktop .deb release? :smile_cat:

**Comment by gohai** [comment]
*Look behind you, a three headed monkey!* \*runs\*

I'll think about it.. I'd do it in a heartbeat if I was running Debian on my own machine, but I unfortunately it's Fedora (.rpm) and OS X.

I agree that it'd be nice to have, but I don't think it's uncommon for projects to only publish tar-balls, plus packaging metadata in the sources, that can be used to automatically generate distribution packages *iff* anyone is sufficiently inclined. But what if the .deb would suddenly not work in some far-out Ubuntu derivative? And what about the distros that don't do .deb?

**Comment by benfry** [comment]
I'm not gonna fall for the three-headed monkey trick again. 

FWIW, the difference for the .deb is that Ubuntu is our only "officially" supported Linux flavor, and it seems like .deb distributions (Ubuntu, Mint, Debian) are most prominent. But obviously, we've also been fine with the tarball to date.

**Comment by GABBAR1947** [comment]
I'd like to work on this. Can anyone help me initiate the work...?

**Comment by kfeuz** [comment]
Just to clarify, the current `ant deb`  build process produces a binary .deb file.  This is great for having an installable .deb file but is nowhere close to what would need to happen to get this into an official repo (so you can apt-get install).  If someone wants to start down that road I'll provide what assistance I can (which isn't much that's why I didn't pursue it).  I am happy to maintain the binary .deb build and can even provide the built .debs to be downloaded.  I can build and test on the current stable release (x86/amd64 only) of debian, ubuntu and mint for each new release and handle any bug reports directly related.  What I am currently lacking is guidance on what to do with the binaries.  It would be nice if they could just be hosted with the other processing downloads but as long as traffic is reasonable (which I imagine it would be) I could always host them on my own site which could be linked to from the processing site.  @benfry any preference?

**Comment by benfry** [comment]
> Just to clarify, the current ant deb build process produces a binary .deb file. This is great for having an installable .deb file but is nowhere close to what would need to happen to get this into an official repo (so you can apt-get install).
Exactly; which is why it's not supported now. Fixing `ant deb` is super easy. Being the maintainer, getting things into apt-get, and maintaining that over a long (multiple releases) period of time is a lot more work.

**Comment by crazymaster** [comment]
>Debian package
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=433270

There is no progress recently.

**Comment by GKFX** [comment]
@benfry I do use Debian as my main OS, and I'd be happy to be the maintainer for it. Can I restart the Debian intent-to-package linked by crazymaster?

**Comment by valerio-bozzolan** [comment]
I see that this is related to #4415.

**Comment by benfry** [comment]
@GKFX By all means.

**Comment by ichnaton** [comment]
What about the status of this ?

My clone currently builds deb, really nice I guess.

**Comment by GKFX** [comment]
@ichnaton Status is that I've made plenty of progress, but there's still more for me to do. The `ant deb` task you are familiar with just isn't suitable for the official Debian repository; you need to submit a "source package" compliant with all their many policy requirements, which then can be used to build the "binary packages" that users download. The `ant deb` task that you are familiar with just builds a non-compliant binary package; my version now builds a mostly-compliant source package and then further Ant tasks are called by the Debian official build tools (within the source package) to make the needed binary packages.

**Comment by OpenSauce04** [comment]
How has this not been touched for nearly 7 years? Having at least a deb package available to download in the releases is a bare minimum option for a project like this
